services/priceService.ts

```diff
@@ -66,25 +65,70 @@ export async function upsertCurrentPricesForNetworkId (responseData: IResponseDa
   await upsertPricesForNetworkId(responseData, networkId, 0)
 }
 
-export async function getPriceForDayTicker (day: moment.Moment, ticker: string, attempt: number = 1): Promise<IResponseData | null> {
+function validateTicker (ticker: string): void {
+  if (!Object.values(TICKERS).includes(ticker)) {
+    throw new Error(RESPONSE_MESSAGES.INVALID_TICKER_400.message)
+  }
+}
+
+function validatePriceAPIUrlAndToken (): void {
   if (appInfo.priceAPIURL === '') {
     throw new Error(RESPONSE_MESSAGES.MISSING_PRICE_API_URL_400.message)
   }
-  if (!Object.values(TICKERS).includes(ticker)) {
-    throw new Error(RESPONSE_MESSAGES.INVALID_TICKER_400.message)
+  if (appInfo.priceAPIToken === '') {
+    throw new Error(RESPONSE_MESSAGES.MISSING_PRICE_API_TOKEN_400.message)
   }
+}
+
+function getPriceByDayURL (day: moment.Moment, ticker: string): string {
+  validatePriceAPIUrlAndToken()
+  validateTicker(ticker)
+  return `${appInfo.priceAPIURL}/pricebydate/${appInfo.priceAPIToken}/${ticker}+${day.format(PRICE_API_DATE_FORMAT)}`
+}
+
+function getAllPricesByTickerURL (ticker: string): string {
+  validatePriceAPIUrlAndToken()
+  validateTicker(ticker)
+  return `${appInfo.priceAPIURL}/dailyprices/${appInfo.priceAPIToken}/${ticker}`
+}
 
-  const dayString = day.format(PRICE_API_DATE_FORMAT)
+export async function getPriceForDayTicker (day: moment.Moment, ticker: string, attempt: number = 1): Promise<IResponseData | null> {
   try {
-    const res = await axios.get(`${appInfo.priceAPIURL}/${ticker}+${dayString}`, {
-      timeout: 30 * 1000
+    const res = await axios.get(getPriceByDayURL(day, ticker), {
+      timeout: PRICE_API_TIMEOUT
     })
 
     if (res.data.success !== false) { return res.data } else { return null }
   } catch (error) {
     console.error(`Problem getting price of ${ticker} ${day.format('YYYYMMDD')} -> ${error as string} (attempt ${attempt})`)
 
-    if (attempt < 5) { return await getPriceForDayTicker(day, ticker, ++attempt) } else { return null }
+    if (attempt < PRICE_API_MAX_RETRIES) {
+      return await getPriceForDayTicker(day, ticker, attempt + 1)
+    } else { return null }
+  }
+}
+
+export async function getAllPricesByTicker (ticker: string, attempt: number = 1): Promise<IResponseDataDaily[] | null> {
+  try {
+    const res = await axios.get(getAllPricesByTickerURL(ticker), {
+      timeout: PRICE_API_TIMEOUT
+    })
+
+    if (res.data.success !== false) {
+      const dailyPrices: IResponseDataDaily[] = Object.entries<IResponseData>(res.data).map(e => {
```

Comment:
Destructure the `e` argument to have more descriptive values than `e[0]` and `e[1]`.
Something like 
```js
.map( ([day, priceData]) => ({ day, ...priceData }) )
```

Reply:
nice, done